callbacks/opentelemetry_test.go

```diff
@@ -0,0 +1,221 @@
+	finish := schema.AgentFinish{}
+
+	handler.HandleAgentFinish(ctx, finish)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleChain(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	tracer := noop.NewTracerProvider().Tracer("test")
+	handler, err := NewOpenTelemetryCallbacksHandler(tracer)
+	require.NoError(t, err)
+
+	inputs := make(map[string]interface{})
+	info := schema.ChainInfo{}
+	next := func(ctx context.Context) (map[string]interface{}, error) {
+		spanCtx := trace.SpanContextFromContext(ctx)
+		assert.NotEmpty(t, spanCtx.SpanID().String())
+		assert.NotEmpty(t, spanCtx.TraceID().String())
+		return map[string]interface{}{
+			"done": "right",
+		}, nil
+	}
+	res, err := handler.HandleChain(ctx, inputs, info, next)
+
+	require.NoError(t, err)
+	assert.Equal(t, "right", res["done"])
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleChainEnd(t *testing.T) {
+	t.Parallel()
+	handler := &OpenTelemetryCallbacksHandler{}
+	ctx := context.Background()
+	outputs := make(map[string]interface{})
+
+	handler.HandleChainEnd(ctx, outputs)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleChainError(t *testing.T) {
+	t.Parallel()
+	handler := &OpenTelemetryCallbacksHandler{}
+	ctx := context.Background()
+	err := assert.AnError
+
+	handler.HandleChainError(ctx, err)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleChainStart(t *testing.T) {
+	t.Parallel()
+	handler := &OpenTelemetryCallbacksHandler{}
+	ctx := context.Background()
+	inputs := make(map[string]interface{})
+
+	handler.HandleChainStart(ctx, inputs)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleLLM(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	tracer := noop.NewTracerProvider().Tracer("test")
+	handler, err := NewOpenTelemetryCallbacksHandler(tracer)
+	require.NoError(t, err)
+
+	messages := []llms.MessageContent{}
+	options := llms.CallOptions{}
+	next := func(ctx context.Context) (*llms.ContentResponse, error) {
+		spanCtx := trace.SpanContextFromContext(ctx)
+		assert.NotEmpty(t, spanCtx.SpanID().String())
+		assert.NotEmpty(t, spanCtx.TraceID().String())
+		return &llms.ContentResponse{
+			Choices: []*llms.ContentChoice{},
+			Usage: &llms.Usage{
+				PromptTokens:     1,
+				CompletionTokens: 1,
+				TotalTokens:      2,
+			},
+		}, nil
+	}
+	res, err := handler.HandleLLM(ctx, messages, options, next)
+
+	require.NoError(t, err)
+	assert.NotNil(t, res)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleLLMError(t *testing.T) {
+	t.Parallel()
+	handler := &OpenTelemetryCallbacksHandler{}
+	ctx := context.Background()
+	err := assert.AnError
+
+	handler.HandleLLMError(ctx, err)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleLLMGenerateContentEnd(t *testing.T) {
+	t.Parallel()
+	handler := &OpenTelemetryCallbacksHandler{}
+	ctx := context.Background()
+	res := &llms.ContentResponse{}
+
+	handler.HandleLLMGenerateContentEnd(ctx, res)
+}
+
+func TestOpenTelemetryCallbacksHandler_HandleLLMGenerateContentStart(t *testing.T) {
```

Comment:
tests like these touch the code, but not sure if we are asserting in a way that's of high value. This is one reason I think we should pare down the code to text completions then make sure all the tests are nice and meaningful. then go back and add things like tools, etc.